src/firestarter/Main.cpp

```diff
@@ -0,0 +1,380 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/\>.
+ *
+ * Contact: daniel.hackenberg@tu-dresden.de
+ *****************************************************************************/
+
+#include <firestarter/Firestarter.hpp>
+#include <firestarter/Logging/Log.hpp>
+
+#include <cxxopts.hpp>
+
+#include <string>
+#include <thread>
+
+int print_copyright(void) {
+
+  firestarter::log::info()
+      << "\n"
+      << "This program is free software: you can redistribute it and/or "
+         "modify\n"
+      << "it under the terms of the GNU General Public License as published "
+         "by\n"
+      << "the Free Software Foundation, either version 3 of the License, or\n"
+      << "(at your option) any later version.\n"
+      << "\n"
+      << "You should have received a copy of the GNU General Public License\n"
+      << "along with this program.  If not, see "
+         "<http://www.gnu.org/licenses/>.\n";
+
+  return EXIT_SUCCESS;
+}
+
+int print_warranty(void) {
+
+  firestarter::log::info()
+      << "\n"
+      << "This program is distributed in the hope that it will be useful,\n"
+      << "but WITHOUT ANY WARRANTY; without even the implied warranty of\n"
+      << "MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n"
+      << "GNU General Public License for more details.\n"
+      << "\n"
+      << "You should have received a copy of the GNU General Public License\n"
+      << "along with this program.  If not, see "
+         "<http://www.gnu.org/licenses/>.\n";
+
+  return EXIT_SUCCESS;
+}
+
+int print_help(cxxopts::Options parser) {
+
+  // clang-format off
+  firestarter::log::info()
+    << parser.help()
+		<< "\n"
+    << "Examples:\n"
+    << "\n"
+    << "  ./FIRESTARTER                 starts FIRESTARTER without timeout\n"
+    << "  ./FIRESTARTER -t 300          starts a 5 minute run of FIRESTARTER\n"
+    << "  ./FIRESTARTER -l 50 -t 600    starts a 10 minute run of FIRESTARTER with\n"
+		<< "                                50\% high load and 50\% idle time\n"
+#ifdef CUDA
+    << "                                on CPUs and full load on GPUs\n"
+#endif
+    << "  ./FIRESTARTER -l 75 -p 20000000\n"
+    << "                                starts FIRESTARTER with an interval length\n"
+    << "                                of 2 sec, 1.5s high load and 0.5s idle\n"
+#ifdef CUDA
+    << "                                on CPUs and full load on GPUs\n"
+#endif
+    ;
+  // clang-format on
+
+  return EXIT_SUCCESS;
+}
+
+int main(int argc, char **argv) {
```

Comment:
command-line option parsing and booting up firestarter should be separated into functions.